packages/flame/lib/src/components/component.dart

```diff
@@ -203,6 +206,17 @@ class Component {
   /// ```
   void onMount() {}
 
+  /// A future that will complete once the component is mounted on its parent
+  Future<void> get hasMounted {
```

Comment:
I don't know why, but `hasMounted` doesn't convince me as the method name. If I ever read this I'll expect to get a boolean value from it not a `Future<void>`. @erickzanardo suggested `mounted` which I somewhat prefer over `hasMounted`.

In addition, I have a question on why do we think that using a `Completer` is better than registering callbacks (for example, `whenMounted(callback)`?

Reply:
I like the `mounted` name! @spydon what do you think?

About the future vs callback, I think it have a nicer API, and actually, by using a future, we can let the user use a callback if they prefer, by doing `component.mounted.wheComplete(fn)`, so we can have the best of both worlds.

Reply:
@erickzanardo true. Missed that API! 

I think it's nice if we have it as `mounted`. I love how it reads:

`component.mounted.whenComplete`
and 
`await component.mounted`